source/Mlos.Python/mlos/Optimizers/long_haul_tests/LongHaulTestBayesianOptimizer.py

```diff
@@ -350,3 +354,41 @@ def run_optimization(optimizer):
             best_config_point, best_objective = optimizer.optimum()
             print(f"Optimum config: {best_config_point}, optimum objective: {best_objective}")
             self.assertLessEqual(sign * best_objective['function_value'], -5.5)
+
+    def validate_optima(self, optimizer):
+        if not optimizer.get_optimizer_convergence_state().surrogate_model_fit_state.fitted:

```

Comment:
Maybe also add ``BEST_VALUE``?

Reply:
Sure, you were already validating that inline, so I thought it was fine to leave those checks untouched and invoke `validate_optima()` afterwards.

Reply:
It might be nice to have them together but I have no strong opinion.

Reply:
I'd much rather refactor our code now instead of the tests.